primitives/src/currency.rs

```diff
@@ -72,3 +72,31 @@ impl TryFrom<CurrencyId> for TokenSymbol {
         }
     }
 }
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash))]
+pub enum PoolAsset {
+    Asset(u32),
+}
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash))]
+pub enum CurrencyOrAsset {
+    NativeCurrency,
+    Asset(u32),
+    PoolAsset(PoolAsset, PoolAsset),
+}
```

Comment:
I don't get why you have to create `PoolAsset` since you could create an asset on the fly in `pallet-assets` which would then work with `CurrencyOrAsset::Asset`.

You'd have a flow like this in the event when you are creating a new pool:
1. Call `pallet-assets`'s `force_create` extrinsic to create a new asset from our own `AccountId`
2. Map our in storage pool state to the asset created

Reply:
@ETeissonniere `pallet-assets`'s `force_create` would work for one asset at a time. We would get two assets `base_asset` and `quote_asset` in our dispatch able functions. 


If I understand your point, we have to create two assets against `base_asset` and `quote_asset`. We would also call `total_issuance` `mint_into` `burn_from` twice for each asset. 

Please correct me if I am wrong. Lets have a call to discuss this.

Reply:
Let's chat on our call. I think there are some confusions